File: 821 - Shorteset Distance to a Character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> list[int]:
        cset = set()
        for i in range(len(s)):
            if s[i] == c:
                cset.add(i)
        
        res = []
        for i in range(len(s)):
            dists = [abs(i - index) for index in cset]
            mindist = min(dists)
            res.append(mindist)
        return res

File: 821 - Shorteset Distance to a Character.js
/**
 * @param {string} s
 * @param {character} c
 * @return {number[]}
 */
var shortestToChar = function(s, c) {
    let res = Array(s.length);
    let prev = -s.length;
    for (let i = 0; i < s.length; i++) {
        if (s[i] == c) {
            prev = i;
        }
        res[i] = i - prev;
    }
    prev = s.length * 2;
    for (let i = s.length - 1; i >= 0; i--) {
        if (s[i] == c) {
            prev = i;
        }
        res[i] = Math.min(res[i], prev - i);
    }
    return res;
};